dashboard/app/assets/javascripts/levels/contract_match.js.jsx

```diff
@@ -1,5 +1,42 @@
 /** TODO(bjordan): Move this all into StudioApps */
 $(window).load(function () {
+  $.widget("custom.coloriconselectmenu", $.ui.selectmenu, {
+    /**
+     * Override the jQuery selectmenu to add a color square icon driven by the
+     * data-color attribute on select elements.
+     * @param ul
+     * @param item
+     * @returns {*}
+     * @private
+     */
+    _renderItem: function (ul, item) {
+      var li = $("<li>", {text: item.label});
+      var color = item.element.attr("data-color");
+      makeColorSquareIcon(color).appendTo(li);
+      return li.appendTo(ul);
+    },
+    styleCurrentValue: function () {
+      styleButton(this.element);
+    }
+  });
+
+  function bgColorStyle(color) {
+    return "background-color: " + color;
+  }
+
+  function styleButton(selectElement) {
+    var selectedColor = $(selectElement).find("option:selected").attr("data-color");
+    var hiddenSelectMenuBtn = "#" + $(selectElement).attr("id") + "-button .ui-selectmenu-text";
```

Comment:
I might put this string inside the $(). The variable name makes me expect to have a button object, not a string that is a selector to get a button object.